### PR TITLE
Blaze: use translated post type name in post-publish panel

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

Instead of relying on the raw post type slug, let's pull the translated post type label. We need to ensure it is lowercase, since it's used in a sentence.

| **Before** | **After** |
|  ---  |  ---  |
|          Screenshot 2023-06-15 at 14 37 03          |        Screenshot 2023-06-15 at 14 40 42         |

> **Note**
> In the French translation above, I had to move the non-translated post type name at the end of the string, where it looked less worse standing on its own. Once this fix is merged and shipped, I will be able to update the translated string again to form a proper sentence without parenthesis.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Go to Settings > General, and switch your site language to French.
* Go to Dashboard > Updates, and update translations if necessary.
* Go to Posts > Add New, write a new post.
* Publish your post.
* View the post-publish panel and its Blaze section; the post type name should be translated.

### CODE REVIEW

**Review by samiff — DISMISSED** [review]

**Comment by jeherve** [comment]
I had to update with latest `trunk` after merging another PR, so this will need another approval I'm afraid.
Thank you!

**Review by samiff — APPROVED** [review]

**Pull request merged**
